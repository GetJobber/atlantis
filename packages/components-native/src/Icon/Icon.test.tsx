import React from "react";
import { render } from "@testing-library/react-native";
import { Icon } from ".";

it("renders home icon", () => {
  const tree = render(<Icon name="home" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders apple icon", () => {
  const tree = render(<Icon name="apple" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders large arrowDown icon", () => {
  const tree = render(<Icon name="arrowDown" size="large" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders thumbsDown icon", () => {
  const tree = render(<Icon name="thumbsDown" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders small more icon", () => {
  const tree = render(<Icon name="more" size="small" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders star icon with custom color", () => {
  const tree = render(<Icon name="star" customColor="#f33323" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders quote icon with themed color", () => {
  // The intention is to make sure the color prop can be applied to an icon with a built-in default color
  const tree = render(<Icon name="quote" color="brand" />).toJSON();
  expect(tree).toMatchSnapshot();
});
