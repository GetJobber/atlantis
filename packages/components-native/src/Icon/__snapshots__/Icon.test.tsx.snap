// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders apple icon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "display": "flex",
        "fill": "hsl(198, 35%, 21%)",
        "height": 24,
        "verticalAlign": "middle",
        "width": 24,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  testID="apple"
  vbHeight={24}
  vbWidth={24}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4280499528,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M17.4507 12.6359C17.4279 10.1092 19.5195 8.87999 19.6152 8.8228C18.4307 7.09577 16.5946 6.85979 15.9494 6.84095C14.4074 6.67867 12.9116 7.76365 12.1261 7.76365C11.3249 7.76365 10.1152 6.85659 8.81151 6.88333C7.13383 6.90917 5.56439 7.88051 4.70358 9.38884C2.92715 12.4643 4.25206 16.9838 5.954 19.4697C6.8054 20.6872 7.80022 22.0466 9.10238 21.9988C10.3763 21.9462 10.8521 21.1866 12.3895 21.1866C13.9127 21.1866 14.3596 21.9988 15.6876 21.9682C17.0548 21.9463 17.9156 20.7453 18.7372 19.5167C19.721 18.1213 20.1162 16.747 20.1318 16.6765C20.0996 16.6655 17.4765 15.6643 17.4507 12.6359ZM14.942 5.20559C15.6272 4.34877 16.096 3.18297 15.9658 2C14.9741 2.04389 13.7339 2.68599 13.0197 3.524C12.3879 4.26242 11.8234 5.47289 11.9692 6.61122C13.0832 6.6943 14.2271 6.04905 14.942 5.20559Z"
      fill={
        {
          "payload": 4280499528,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders home icon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "display": "flex",
        "fill": "hsl(198, 35%, 21%)",
        "height": 24,
        "verticalAlign": "middle",
        "width": 24,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  testID="home"
  vbHeight={24}
  vbWidth={24}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4280499528,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M13.147 3.582a2 2 0 0 0-2.294 0l-9.426 6.599a1 1 0 1 0 1.147 1.638L5 10.121V18a3 3 0 0 0 3 3h8a3 3 0 0 0 3-3v-7.88l2.427 1.7a1 1 0 0 0 1.146-1.64l-9.426-6.598ZM17 8.721V18a1 1 0 0 1-1 1H8a1 1 0 0 1-1-1V8.72l5-3.5 5 3.5Z"
      fill={
        {
          "payload": 4280499528,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders large arrowDown icon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={32}
  bbWidth={32}
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "display": "flex",
        "fill": "hsl(198, 35%, 21%)",
        "height": 32,
        "verticalAlign": "middle",
        "width": 32,
      },
      {
        "flex": 0,
        "height": 32,
        "width": 32,
      },
    ]
  }
  testID="arrowDown"
  vbHeight={24}
  vbWidth={24}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4280499528,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M7.703 8.291a.996.996 0 0 0-1.41.001.994.994 0 0 0 0 1.41l5 5.005a.998.998 0 0 0 1.415 0l5-5a.994.994 0 0 0 0-1.41.995.995 0 0 0-1.411-.001L12 12.584 7.703 8.291Z"
      fill={
        {
          "payload": 4280499528,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders quote icon with themed color 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "display": "flex",
        "fill": "hsl(79, 85%, 34%)",
        "height": 24,
        "verticalAlign": "middle",
        "width": 24,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  testID="quote"
  vbHeight={24}
  vbWidth={24}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4285702157,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M14 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Zm-2 0a1 1 0 1 0-2 0 1 1 0 0 0 2 0Z"
      fill={
        {
          "payload": 4285702157,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M3.586 6C3.196 6.39 3 6.902 3 7.414c0 .512.195 1.024.586 1.414l.99.991C4.315 10.554 4 11.175 4 12a7 7 0 0 0 7 7h9v1a1 1 0 1 0 2 0v-2a1 1 0 0 0-1-1h-3v-5a7 7 0 0 0-7-7c-.825 0-1.446.314-2.18.577l-.992-.991A1.994 1.994 0 0 0 6.414 4c-.512 0-1.023.195-1.414.586L3.586 6Zm3.393.565A7.041 7.041 0 0 0 5.565 7.98L5 7.414 6.414 6l.565.565ZM11 7a5 5 0 0 1 5 5v5h-5a5 5 0 0 1 0-10Z"
      fill={
        {
          "payload": 4285702157,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders small more icon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={16}
  bbWidth={16}
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "display": "flex",
        "fill": "hsl(198, 35%, 21%)",
        "height": 16,
        "verticalAlign": "middle",
        "width": 16,
      },
      {
        "flex": 0,
        "height": 16,
        "width": 16,
      },
    ]
  }
  testID="more"
  vbHeight={24}
  vbWidth={24}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4280499528,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M4 12a2 2 0 1 1 4 0 2 2 0 0 1-4 0Zm6 0a2 2 0 1 1 4 0 2 2 0 0 1-4 0Zm8-2a2 2 0 1 0 0 4 2 2 0 0 0 0-4Z"
      fill={
        {
          "payload": 4280499528,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders star icon with custom color 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "display": "flex",
        "fill": "hsl(198, 35%, 21%)",
        "height": 24,
        "verticalAlign": "middle",
        "width": 24,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  testID="star"
  vbHeight={24}
  vbWidth={24}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4280499528,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="m15.673 13.337 3.673-3.58-5.076-.738-2.27-4.6-2.27 4.6-5.076.738 3.673 3.58-.867 5.176L12 16.006l4.54 2.386-.867-5.055ZM12 18.335l-4.505 2.49a1.546 1.546 0 0 1-2.244-1.63l.86-5.137-3.644-3.553a1.546 1.546 0 0 1 .857-2.637l5.037-.732 2.252-4.273a1.547 1.547 0 0 1 2.774 0l2.252 4.273 5.037.732a1.546 1.546 0 0 1 .857 2.637l-3.645 3.553.86 5.139a1.547 1.547 0 0 1-2.243 1.63L12 18.334Z"
      fill={
        {
          "payload": 4294128419,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders thumbsDown icon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "display": "flex",
        "fill": "hsl(198, 35%, 21%)",
        "height": 24,
        "transform": [
          {
            "scaleY": -1,
          },
        ],
        "verticalAlign": "middle",
        "width": 24,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  testID="thumbsDown"
  transform={
    [
      {
        "scaleY": -1,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4280499528,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M8 11.78V19h8l3-5.76v-1.29h-7.639l1.036-7.944L8 11.78ZM6.134 11l4.555-8.034c1.124-1.846 3.971-.844 3.691 1.299l-.74 5.685h6.86a.5.5 0 0 1 .5.5v3a1 1 0 0 1-.084.4l-3.276 6.648a.95.95 0 0 1-.44.453c-.132.065-.276.049-.423.049H4c-1.105 0-2-1.045-2-2.15V13a2 2 0 0 1 2-2h2.134ZM6 13H4v6h2v-6Z"
      fill={
        {
          "payload": 4280499528,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
